### PR TITLE
samples:openthread: use nrf_oberon for openthread samples

### PR DESCRIPTION
**Pull request by piotrkoziar**
Replaced default mbedtls with oberon on nRF52833. This makes
    commissioning work a lot faster.
    
    Signed-off-by: Piotr Koziar <piotrkoziar@hotmail.com>
    Signed-off-by: Marek Porwisz <marek.porwisz@nordicsemi.no>

### CODE REVIEW

**Comment by piotrkoziar** [comment]
Work in progress

File: subsys/net/openthread/Kconfig [hunk 1]
```diff
@@ -13,14 +13,12 @@ config OPENTHREAD_NRF_SECURITY
 	depends on !OPENTHREAD_MBEDTLS
 	select NORDIC_SECURITY_BACKEND
 	select OBERON_BACKEND
-	select MBEDTLS_VANILLA_BACKEND
```
Comment (tejlmand) [inline subsys/net/openthread/Kconfig:16]: Nothing wrong, but what is the reasons behind this Kconfig file, and the one here:
https://github.com/nrfconnect/sdk-nrfxlib/blob/master/openthread/Kconfig

Relates to this comment:
https://github.com/nrfconnect/sdk-nrfxlib/pull/244#discussion_r480937765

Could you try to align those two PRs and in each Kconfig file (if both are required), then make a description in each file on what kind of configuration should go where.

**Review by lmaciejonczyk — APPROVED** [review]

File: subsys/net/openthread/Kconfig (hunk 1 above)
Comment (tejlmand) [reply subsys/net/openthread/Kconfig:16]: @piotrkoziar still waiting for an answer ?

Comment (MarekPorwisz) [reply subsys/net/openthread/Kconfig:16]: There are three OpenThread Kconfig files in three repos:
- zephyr - Primary file related to the OT sources in zephyr
- nrf - exists to extend/override some settings that upstream zephyr should not be aware of. Settings applicable for building from sources or common to both sources and libs
- nrfxlib - settings related to openthread precompiled libraries only.

Comment (MarekPorwisz) [reply subsys/net/openthread/Kconfig:16]: added following description to the file:
# The purpose of this file is to override and extend settings related to
# building Thread from sources. It extends the zephyrs openthread Kconfig file.
# Separate Kconfig exists in nrfxlib repository responsible solely for managing
# OpenThread precompiled libraries.

**Comment by lmaciejonczyk** [comment]
@MarekPorwisz @rlubos @tejlmand Are you ok with the changes? Can we merge it finally?

**Review by rlubos — APPROVED** [review]

**Review by MarekPorwisz — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]

**Pull request merged**